edge/pkg/edgehub/edgehub_test.go

```diff
@@ -1,95 +1,302 @@
-			t.Errorf("Group of module is not correct wanted: %v and got: %v", commodule.HubGroup, edgeHubModule.Group())
+		if _, ok := certSync["metamanager"]; !ok {
+			t.Error("NewCertSyncChannel() returned  unexpected results. expected key metamanager to be present but it was not available.")
+
 		}
 	})
 }
 
-//TestStart is a function to test the start of the edge hub module
-func TestStart(t *testing.T) {
-	// time to let config be synced again
-	time.Sleep(10 * time.Second)
-	coreContext = context.GetContext(context.MsgCtxTypeChannel)
-	modules := core.GetModules()
-	for name, module := range modules {
-		coreContext.AddModule(name)
-		coreContext.AddModuleGroup(name, module.Group())
+func TestRegister(t *testing.T) {
+	tests := []struct {
+		eh           *v1alpha2.EdgeHub
+		nodeName     string
+		name         string
+		wantNodeName string
+	}{
+		{
+			name:         "",
+			nodeName:     "test1",
+			wantNodeName: "test1",
+			eh:           &v1alpha2.EdgeHub{WebSocket: &v1alpha2.EdgeHubWebSocket{Server: "localhost:8080"}, ProjectID: "test_id"},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			Register(tt.eh, tt.nodeName)
+
+			if config.Config.NodeName != tt.wantNodeName {
+				t.Errorf("failed to Register(). Nodename : got = %s, want = %s", config.Config.NodeName, tt.wantNodeName)
+			}
+
+		})
 	}
-	go edgeHubModule.Start(coreContext)
-	time.Sleep(2 * time.Second)
 }
 
-// TestCleanup is function to test cleanup
-func TestCleanup(t *testing.T) {
-	edgeHubModule.Cleanup()
-	var test model.Message
+func TestName(t *testing.T) {
+	t.Run("EdgeHub.Name()", func(t *testing.T) {
+		if got := (&EdgeHub{}).Name(); got != "websocket" {
+			t.Errorf("EdgeHub.Name() returned unexpected result. got = %s, want = websocket", got)
+		}
 
-	// Send message to avoid deadlock if channel deletion has failed after cleanup
-	go coreContext.Send(ModuleNameEdgeHub, test)
+	})
+}
 
-	_, err := coreContext.Receive(ModuleNameEdgeHub)
-	t.Run("CheckCleanUp", func(t *testing.T) {
-		if err == nil {
-			t.Errorf("Edgehub Module still has channel after cleanup")
+func TestGroup(t *testing.T) {
+	t.Run("EdgeHub.Group()", func(t *testing.T) {
+		if got := (&EdgeHub{}).Group(); got != "hub" {
+			t.Errorf("EdgeHub.Group() returned unexpected result. got = %s, want = hub", got)
 		}
+
 	})
 }
 
-*/
+func TestEnable(t *testing.T) {
+	tests := []struct {
+		eh   *EdgeHub
+		want bool
+		name string
+	}{
+		{
+			name: "Enable true",
+			want: true,
+			eh:   &EdgeHub{enable: true},
+		},
+		{
+			name: "Enable false",
+			want: false,
+			eh:   &EdgeHub{enable: false},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if got := tt.eh.Enable(); got != tt.want {
+				t.Errorf("EdgeHub.Enable() returned expected results. got = %v, want = %v", got, tt.want)
+			}
+		})
+	}
+}
+
+// TODO: eh.Start will wait until we call beehiveContext.Cancel(). When running `make test` tests, it seems that
```

Comment:
Hi Team,
I'm was not able to add test Start() function. 

Inside the `Start()`, We wait for `beehiveContext.Done() channel.  The context done is channel is a global variable (initialised by init() function). It always blocks the `testing.Run()` forever. (I think this is due to nil context.Done() channel)

If I run only one test(There are two testcases 1. Websocket  2.Quic), then it'll work.  If I run the both test cases, then the sample problem will happen.

Reply:
Thanks!
I'd like to double check this block.

Reply:
@ksankeerth I tested on my local host, we can try to new a mock client same as here https://github.com/kubeedge/kubeedge/blob/master/edge/pkg/edgehub/process_test.go#L126.
That would be well to do the unit test.

Reply:
Thanks for the suggestion. I'll update the PR  👍

Reply:
I wrote tests for `EdgeHub.Start()` and seems to be working. Please check. Thanks